### PR TITLE
Add "go vet" validation to check program bug

### PR DESCRIPTION
**Pull request by WeiZhang555**
Add go vet validation script and fix some vet bugs.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

### CODE REVIEW

**Comment by WeiZhang555** [comment]
Try to use this script to check go vet error, you can find:
```
$ bash validate-vet.sh 
Errors from go vet:
 - hypervisor/network/network_linux.go:335: wrong number of args for format in Errorf call: 1 needed but 2 args
hypervisor/network/network_linux.go:1032: arg tapIface for printf verb %s of wrong type: *net.Interface
hypervisor/network/network_linux.go:1136: arg tapIface for printf verb %s of wrong type: *net.Interface
exit status 1
 - hypervisor/libvirt/libvirt.go:228: struct field tag `xml:"maxMemory",omitempty` not compatible with reflect.StructTag.Get: bad syntax for struct tag pair
hypervisor/libvirt/libvirt.go:549: struct field tag `xml:type,attr` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
hypervisor/libvirt/libvirt.go:590: struct field tag `xml: "name,auth,omitempty"` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
hypervisor/libvirt/libvirt.go:703: missing argument for Errorf("%d"): format reads arg 1, have only 0 args
exit status 1

Please fix the above errors. You can test via "go vet" and commit the result.

```

Adding this script to integration test will be a better choice, sadly I'm not sure how :cry:

**Comment by WeiZhang555** [comment]
After fix related go vet errors:
```
$ ./hack/validate-vet.sh 
Congratulations!  All Go source files have been vetted.
```

**Comment by feiskyer** [comment]
@WeiZhang555 Could you add `./hack/validate-vet.sh` to `.travis.yml` (after `hack/verify-gofmt.sh`) ?

**Comment by WeiZhang555** [comment]
Done. :wink: 

Another validation we can consider is  [golint](https://github.com/golang/lint) which is also applied to `docker`, it can help make code cleaner though not all people like its code style. But this can cost lots of time, we can do it later.

File: hack/validate-vet.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        for err in "${errors[@]}"; do
+            echo " - $err"
+        done
+        echo
+        echo 'Please fix the above errors. You can test via "go vet" and commit the result.'
+        echo
+    } >&2 
+    false
+fi
+
```
Comment (laijs) [inline hack/validate-vet.sh]: it contains several trailing spaces, please remove it.

Comment (WeiZhang555) [inline hack/validate-vet.sh]: Done

**Comment by feiskyer** [comment]
LGTM

**Comment by laijs** [comment]
it still contains trailing space.

**Comment by WeiZhang555** [comment]
I checked but didn't find it. It's strange :question:

**Comment by WeiZhang555** [comment]
Do you mean the `starting` spaces? I can't see the trailing spaces...

**Pull request merged**